src/bootstrap/doc.rs

```diff
@@ -353,7 +353,8 @@ impl Step for Standalone {
         let version_input = builder.src.join("src/doc/version_info.html.template");
         let version_info = out.join("version_info.html");
 
-        if !builder.config.dry_run && !up_to_date(&version_input, &version_info) {
+        if (builder.config.dry_run || up_to_date(&version_input, &version_info)) &&
```

Comment:
It seems like this is wrong. We shouldn't need to generate this file during dry_run; if there's something that reads from it during dry_run then *that* should be gated, not the file creation.